File: src/reflex.c
#include "../include/bcind_kernel.h"

void execute_reflex_action(const char* reason) {
    fprintf(stderr, "\n========================================================\n");
    fprintf(stderr, "   CRITICAL FAIL-SAFE REFLEX ACTION TRIGGERED\n");
    fprintf(stderr, "   Reason: %s\n", reason ? reason : "Unspecified System Boundary Fault");
    fprintf(stderr, "   Status: Pipeline Latched / Output Terminated\n");
    fprintf(stderr, "========================================================\n\n");
}

File: src/reflex.rs
pub fn execute_reflex_action(reason: &str) {
    log::error!("\n========================================================");
    log::error!("   CRITICAL FAIL-SAFE REFLEX ACTION TRIGGERED");
    log::error!(
        "   Reason: {}",
        if reason.is_empty() {
            "Unspecified System Boundary Fault"
        } else {
            reason
        }
    );
    log::error!("   Status: Pipeline Latched / Output Terminated");
    log::error!("========================================================\n");
}
